catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/BackupCommandTest.java

```diff
@@ -0,0 +1,976 @@
+        } else {
+            System.clearProperty("https.protocols");
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        if (miniSolrCloud != null) {
+            miniSolrCloud.getSolrClient()
+                    .close();
+            miniSolrCloud.shutdown();
+        }
+    }
+
+    @Test
+    public void testNoArgBackup() throws Exception {
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_OK,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup of [%s] complete.", DEFAULT_CORE_NAME)));
+
+    }
+
+    @Test
+    public void testBackupSpecificCore() throws Exception {
+        final String coreName = "core";
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_OK,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+
+        backupCommand.coreName = coreName;
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup of [%s] complete.", coreName)));
+
+    }
+
+    @Test
+    public void testBackupInvalidCore() throws Exception {
+        final String coreName = "badCoreName";
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_NOT_FOUND,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+
+        backupCommand.coreName = coreName;
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup command failed due to: %d",
+                        HttpStatus.SC_NOT_FOUND)));
+    }
+
+    @Test
+    public void testSingleNodeBackupAsyncOptionSupplied() throws Exception {
+
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage(SEE_COMMAND_USAGE_MESSAGE);
+
+        when(mockHttpWrapper.execute(any(URI.class))).thenReturn(mockResponse(HttpStatus.SC_OK,
+                ""));
+
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            protected HttpWrapper getHttpClient() {
+                return mockHttpWrapper;
+            }
+        };
+        backupCommand.asyncBackup = true;
+
+        backupCommand.doExecute();
```

Comment:
Leaving as-is since this is an existing test and was not added/modified for this ticket.